src/test/scala/com/databricks/sparkdl/DeepImageFeaturizerSuite.scala

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2017 Databricks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.databricks.sparkdl
+
+import org.apache.spark.image.ImageSchema
+import org.apache.spark.sql.functions.{col, lit}
+import org.apache.spark.ml.linalg.{Vector, Vectors}
+import org.apache.spark.sql.Row
+import org.apache.spark.sql.types.{StructField, StructType}
+import org.scalatest.FunSuite
+
+class DeepImageFeaturizerSuite extends FunSuite with TestSparkContext {
+  test ("Test named image featurizer runs on our image samples"){
+    val outputColName = "myOutput"
+    val featurizer = new DeepImageFeaturizer()
+      .setInputCol("myInput")
+      .setOutputCol(outputColName)
+    val imageDir = getClass.getResource("/images").getFile
+    val data = ImageSchema.readImages(imageDir)
+      .withColumn("myInput", col("image"))
+    val transformed = featurizer.transform(data)
+
+    assert(transformed.columns contains outputColName, "The expected output column was not created.")
+
+    // check that we can materialize a row, and the type is Vector.
+    val vect = transformed.select(col(outputColName)).first().getAs[Vector](0)
+
+    assert(featurizer.transformSchema(data.schema).fieldNames === transformed.columns)
+    assert(featurizer.transformSchema(data.schema) === transformed.schema)
+  }
+
+  test ("Test schema validation.") {
+    val missingInputColumn = "missingInputColumn"
+    val outputColumn = "outputColumn"
+    val featurizer = new DeepImageFeaturizer()
+      .setInputCol(missingInputColumn)
+      .setOutputCol(outputColumn)
+
+    import spark.implicits._
+    val data = sqlContext.createDataset(0 until 100).toDF("columnName")
+
+    assertThrows[IllegalArgumentException] {
+      featurizer.transformSchema(data.schema)
+    }
+
+    assertThrows[IllegalArgumentException] {
+      val hasColumnWithWrongType = data.withColumn(missingInputColumn, lit("str"))
+      featurizer.transformSchema(hasColumnWithWrongType.schema)
+    }
+
+    assertThrows[IllegalArgumentException] {
+      val hasOutputColumn = data.withColumn(outputColumn, lit("str"))
+      featurizer.transformSchema(hasOutputColumn.schema)
+    }
+  }
+
+  test("Test test_net on a known data sample.") {
```

Comment:
do you need `"Test named image featurizer runs on our image samples"` given this test?

Reply:
The "out image samples" test is more of an integration test, it ensures that our featurizer works with a data frame created by "readImages" and makes sure that in can handle images of different sizes and shapes. I can take it out if you think it's too much.

Reply:
ah ok. maybe rename it to "Test named image featurizer runs on image dataframes" ?